app/models/pipeline_output.rb

```diff
@@ -19,4 +19,21 @@ def generate_report
 
     end
   end
+
+  def generate_aggregate_counts(tax_level_name)
+    current_date = Time.zone.now.strftime("%Y-%m-%d")
+    tax_level_id = TaxonCount::NAME_2_LEVEL[tax_level_name]
+    TaxonCount.connection.execute(
+      "INSERT INTO taxon_counts(pipeline_output_id, tax_id, name,
+                                tax_level, count_type, count, created_at, updated_at)
+       SELECT #{id}, taxon_lineages.#{tax_level_name}_taxid, taxon_lineages.#{tax_level_name}_name,
+              #{tax_level_id}, taxon_counts.count_type,
+              sum(taxon_counts.count), '#{current_date}', '#{current_date}'
+       FROM  taxon_lineages, taxon_counts
+       WHERE taxon_lineages.taxid = taxon_counts.tax_id AND
+             taxon_counts.pipeline_output_id = #{id} AND
+             taxon_counts.tax_level = #{TaxonCount::TAX_LEVEL_SPECIES}
+      GROUP BY 1,2,3,4,5"
+    )
+  end
```

Comment:
How does this deal with reads that map to a genus without mapping to any particular species?

Reply:
I don't think our pipeline currently allows for such a case, but it should in the future

Reply:
Ok, then all I request is a small comment here to that effect.

Reply:
I think it already does that for most part because we do have all the taxon ids and some of them do map to 999. I will add a to-do and make the change after we finalize the taxon_leages not found code.